src/store/DrawToolsStore.ts

```diff
@@ -364,15 +363,33 @@ export default class DrawToolsStore {
             activeTools.push({
                 id: finalItem.id,
                 items: [finalItem],
-            });
-        } else {
+        });
+    } else {
             const item = activeTools[groupIndex];
             item.items.push({ ...finalItem, ...{ num: item.items.length } });
             activeTools[groupIndex] = item;
         }
         this.activeToolsGroup = activeTools;
     }
 
+
+    saveTemplate(name:string){
```

Comment:
```suggestion
    saveTemplate(name: string){
```
is a space supposed to be there? same for the below occurrences